modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/40_dhcp.js

```diff
@@ -12,29 +13,82 @@ var callLuciDHCPLeases = rpc.declare({
 return baseclass.extend({
 	title: '',
 
+	isMACStatic: {},
+	isDUIDStatic: {},
+
 	load: function() {
 		return Promise.all([
 			callLuciDHCPLeases(),
-			network.getHostHints()
+			network.getHostHints(),
+			uci.load('dhcp')
 		]);
 	},
 
+	handleCreateStaticLease: function(lease, ev) {
+		ev.currentTarget.classList.add('spinning');
+		ev.currentTarget.disabled = true;
+		ev.currentTarget.blur();
+
+		var cfg = uci.add('dhcp', 'host');
+		uci.set('dhcp', cfg, 'name', lease.hostname || '-');
+		uci.set('dhcp', cfg, 'ip', lease.ipaddr);
+		uci.set('dhcp', cfg, 'mac', lease.macaddr.toUpperCase());
+
+		return uci.save()
+			.then(L.bind(L.ui.changes.init, L.ui.changes))
+			.then(L.bind(L.ui.changes.displayChanges, L.ui.changes));
+	},
+
+	handleCreateStaticLease6: function(lease, ev) {
+		ev.currentTarget.classList.add('spinning');
+		ev.currentTarget.disabled = true;
+		ev.currentTarget.blur();
+
+		var cfg = uci.add('dhcp', 'host');
+		uci.set('dhcp', cfg, 'name', lease.hostname || '-');
+		uci.set('dhcp', cfg, 'duid', lease.duid.toUpperCase());
+		uci.set('dhcp', cfg, 'mac', lease.macaddr);
+
+		return uci.save()
+			.then(L.bind(L.ui.changes.init, L.ui.changes))
+			.then(L.bind(L.ui.changes.displayChanges, L.ui.changes));
+	},
+
 	renderLeases: function(data) {
 		var leases = Array.isArray(data[0].dhcp_leases) ? data[0].dhcp_leases : [],
 		    leases6 = Array.isArray(data[0].dhcp6_leases) ? data[0].dhcp6_leases : [],
-		    machints = data[1].getMACHints(false);
+		    machints = data[1].getMACHints(false),
+		    dhcp_conf = this.uci.state.values.dhcp;
+
+		for (var element in dhcp_conf) {
+
+			if (dhcp_conf[element]['.type'] == 'host') {
+				if (dhcp_conf[element].mac) {
+					var macs = dhcp_conf[element].mac.split(',')
```

Comment:
As I wrote earlier, please use `L.toArray(host.mac)` since the value may be either a whitespace separated string or an array in case uci list notation was used. Using `split()` is not correct in either case.

Also please add a semicolon to the end of the statement.